Artifacts/windows-ConfigurationManagerCurrentBranch(EVAL)/Reset-SqlAdmin.psm1

```diff
@@ -0,0 +1,338 @@
+	  
+	  Tested on Windows XP, 7, 8.1, Server 2012 and Windows Server Technical Preview 2.
+	  Tested on SQL Server 2005 SP4 through 2016 CTP2.
+	  
+	  THIS CODE IS PROVIDED "AS IS", WITH NO WARRANTIES.
+		
+	 .PARAMETER SqlServer
+	  The SQL Server instance. SQL Server must be 2005 and above, and can be a clustered or stand-alone instance.
+	  
+	 .PARAMETER Login
+	  By default, the Login parameter is "sa" but any other SQL or Windows account can be specified. If a login does not
+	  currently exist, it will be added.
+	  
+	  When adding a Windows login to remote servers, ensure the SQL Server can add the login (ie, don't add WORKSTATION\Admin to remoteserver\instance. Domain users and 
+	  Groups are valid input.
+	 
+	 .NOTES 
+	  Author: Chrissy LeMaire
+	  Requires: Admin access to server (not SQL Services), 
+				Remoting must be enabled and accessible if $sqlserver is not local
+	  DateUpdated: 2015-June-06
+	  Version: 0.8
+
+	 .LINK 
+	 https://gallery.technet.microsoft.com/scriptcenter/Reset-SQL-SA-Password-15fb488d
+
+	 .EXAMPLE   
+	 Reset-SqlAdmin -SqlServer sqlcluster
+	 
+	 Prompts for password, then resets the "sa" account password on sqlcluster.
+	 
+	 .EXAMPLE   
+	 Reset-SqlAdmin -SqlServer sqlserver\sqlexpress -Login ad\administrator
+	 
+	 Adds the domain account "ad\administrator" as a sysadmin to the SQL instance. 
+	 If the account already exists, it will be added to the sysadmin role.
+
+	 .EXAMPLE   
+	 Reset-SqlAdmin -SqlServer sqlserver\sqlexpress -Login sqladmin
+	
+	 Prompts for passsword, then adds a SQL Login "sqladmin" with sysadmin privleges. 
+	 If the account already exists, it will be added to the sysadmin role and the password will be reset.
+	
+	 #> 
+			[CmdletBinding()]
+			param(
+				[Parameter(Mandatory = $true)]
+				[string]$SqlServer,
+				[string]$Login = "sa"
+			)
+			
+	BEGIN {
+		Function ConvertTo-PlainText {
+		<#
+			.SYNOPSIS
+			Converts a SecureString to plain text
+
+			.EXAMPLE
+			ConvertTo-PlainText $password
+			
+			.OUTPUT
+			String
+					
+		 #>
+				[CmdletBinding()]
+				param(
+					[Parameter(Mandatory = $true)]
+					[Security.SecureString]$Password
+				)
+				
+			$marshal = [Runtime.InteropServices.Marshal]
+			$plaintext = $marshal::PtrToStringAuto( $marshal::SecureStringToBSTR($Password) )
+			return $plaintext
+		}
+		
+		Function Invoke-ResetSqlCmd {
+		<#
+			.SYNOPSIS
+			Executes a SQL statement against specified computer, and uses "Reset-SqlAdmin" as the
+			Application Name.
+
+			.EXAMPLE
+			Invoke-ResetSqlCmd -sqlserver $sqlserver -sql $sql
+			
+			.OUTPUT
+			$true or $false
+					
+		 #>
+				[CmdletBinding()]
+				param(
+					[Parameter(Mandatory = $true)]
+					[string]$sqlserver,
+					[string]$sql
+				)
+			try { 
+				$connstring = "Data Source=$sqlserver;Integrated Security=True;Connect Timeout=2;Application Name=Reset-SqlAdmin"
```

Comment:
SQL Server 2016 base Image from the public Image Gallery is in Windows Authentication mode and SA is anyway unknown... the SYSTEM Account was removed from the SQL Sysadmins. That's why all Artifacts running as SYSTEM will have to "hack" SQL to get back SysAdmin rights...